sandbox/cgroup/memory.go

```diff
@@ -51,20 +75,38 @@ func CheckMemoryCgroup() error {
 			memoryCgroupValues := strings.Fields(line)
 			if len(memoryCgroupValues) < 4 {
 				// change in size, should investigate the new structure
-				return fmt.Errorf("cannot parse cgroups file: invalid line %q", line)
+				return false, fmt.Errorf("cannot parse cgroups file: invalid line %q", line)
 			}
 			isMemoryEnabled := memoryCgroupValues[3] == "1"
-			if !isMemoryEnabled {
-				return fmt.Errorf("memory cgroup is disabled on this system")
-			}
-			return nil
+			return isMemoryEnabled, nil
 		}
 	}
 
 	if err := scanner.Err(); err != nil {
-		return fmt.Errorf("cannot read %s contents: %v", cgroupsFilePath, err)
+		return false, fmt.Errorf("cannot read %s contents: %v", cgroupsFilePath, err)
 	}
 
-	// no errors so far but the only path here is the cgroups file without the memory line
-	return fmt.Errorf("cannot find memory cgroup in %s", cgroupsFilePath)
+	return false, nil
+}
+
+func checkV2CgroupMemoryController() (bool, error) {
+	// check at the root controller
+	f, err := os.Open(filepath.Join(rootPath, cgroupMountPoint, "cgroup.controllers"))
+	if err != nil {
+		return false, err
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	// expecting a single line
+	if !scanner.Scan() {
+		if err := scanner.Err(); err != nil {
+			return false, err
+		}
+		return false, nil
+	}
+
+	line := scanner.Text()
+	controllers := strings.Fields(line)
+	return strutil.ListContains(controllers, "memory"), nil
```

Comment:
Maybe something like this? Doesn't matter too much though.
```
scanner := bufio.NewScanner(f)
scanner.Split(bufio.ScanWords)
                                      
for scanner.Scan() {
	if scanner.Text() == "memory" {
		return true, nil
	}
}
                                      
if err := scanner.Err(); err != nil {
	return false, err
}
                                      
return false, nil
```

Reply:
good idea, applied